internal/tls/config.go

```diff
@@ -100,6 +91,15 @@ func tlsClientConfigFromSecret(secret corev1.Secret, url string, kubernetesTLSKe
 			secret.Name)
 	}
 
+	// Only Secrets of type Opaque and TLS are allowed. We also allow Secrets with a blank
+	// type, to avoid having to specify the type of the Secret for every test case.
+	// Since a real Kubernetes Secret is of type Opaque by default, its safe to allow this.
+	switch secret.Type {
+	case corev1.SecretTypeOpaque, corev1.SecretTypeTLS, "":
+	default:
+		return nil, nil, fmt.Errorf("cannot use secret '%s' to construct TLS config: invalid secret type: '%s'", secret.Name, secret.Type)
+	}
+
```

Comment:
Isn't the result here the same as before only that we return an error later in the program flow now?

Reply:
the issue arises from SC doing a check for a Secret type while reading TLS data when there's no need to do the check since the Secret doesn't contain any TLS data.

Reply:
My point is that even if the Secret type is `kubernetes.io/dockerconfigjson` this function would still return an error even if the secret contained TLS data.